posthog/api/test/test_personal_api_keys.py

```diff
@@ -440,6 +442,23 @@ def test_works_with_routes_missing_action(self):
         )
         assert response.status_code == status.HTTP_200_OK
 
+    def test_works_with_session_recording_snapshots(self):
+        session_id = str(uuid.uuid4())
+        SessionRecording.objects.create(
+            team=self.team,
+            session_id=session_id,
+            deleted=False,
+            storage_version="2023-08-01",
+            object_storage_path="an lts stored object path",
+        )
+
+        self.key.scopes = ["session_recording:read"]
+        self.key.save()
+        response = self.client.patch(
+            f"/api/projects/{self.team.id}/session_recordings/{session_id}/snapshots?version=2"
```

Comment:
Generally as to whether this makes sense - sure - "I guess so". The reason we were cagey about it before is this API has changed _a lot_ so we don't want to make it public unless we are really happy to support the format basically forever.

Reply:
so this API should not be public hence even full token permission does not work?
What I am looking after is: hey this should work but we have a permission bug or hey this is not a public API (I can fix the docs if that's the case) and follow up asking their use case to call the API directly.